src/status_im/ethereum/subscriptions.cljs

```diff
@@ -37,22 +79,117 @@
+                                           :method  "eth_subscribeSignal"
+                                           :params  [filter params]}))]
+    (status/call-private-rpc
+     args
+     (fn [response]
+       (if (string/blank? response)
+         (log/error ::subscription-unknown-error :filter filter :params params)
+         (let [{:keys [error result]}
+               (-> (.parse js/JSON response)
+                   (js->clj :keywordize-keys true))]
+           (if error
+             (log/error ::subscription-error error :filter filter :params params)
+             (re-frame/dispatch [:ethereum.callback/subscription-success
+                                 result
+                                 callback]))))))))
+
+(defn- add-padding [address]
+  {:pre [(string? address)]}
+  (str "0x000000000000000000000000" (subs address 2)))
+
+(defn- remove-padding [topic]
+  {:pre [(string? topic)]}
+  (str "0x" (subs topic 26)))
+
+(defn- parse-transaction-entries [timestamp chain-tokens direction transfers]
+  {:pre [(integer? timestamp)
+         (map? chain-tokens)
+         (every? (fn [[k v]] (and (string? k) (map? v))) chain-tokens)
+         (keyword? direction)
+         (every? map? transfers)]}
+  (into {}
+        (keep identity
+              (for [transfer transfers]
+                (when-let [token (->> transfer :address (get chain-tokens))]
+                  (when-not (:nft? token)
+                    [(:transactionHash transfer)
+                     {:block         (str (-> transfer :blockNumber ethereum/hex->bignumber))
+                      :hash          (:transactionHash transfer)
+                      :symbol        (:symbol token)
+                      :from          (some-> transfer :topics second remove-padding)
+                      :to            (some-> transfer :topics last remove-padding)
+                      :value         (-> transfer :data ethereum/hex->bignumber)
+                      :type          direction
+                      :gas-price     nil
+                      :nonce         nil
+                      :data          nil
+                      :gas-limit     nil
+                      :timestamp     (str (* timestamp 1000))
+                      :gas-used      nil
+                      ;; NOTE(goranjovic) - metadata on the type of token: contains name, symbol, decimas, address.
+                      :token         token
+                      ;; NOTE(goranjovic) - if an event has been emitted, we can say there was no error
+                      :error?        false
+                      ;; NOTE(goranjovic) - just a flag we need when we merge this entry with the existing entry in
+                      ;; the app, e.g. transaction info with gas details, or a previous transfer entry with old
+                      ;; confirmations count.
+                      :transfer      true}]))))))
+
+(letfn [(combine-entries [transaction token-transfer]
```

Comment:
There are only 2 uses of `letfn` across codebase. In this case we are not using special features of `letfn` (ability to reference each other in all bindings). Maybe makes sense to replace it with let inside the body of of `dedupe-transactions`. Also, `both-transfer?` is not used anywhere.

Reply:
I will be replaced in next PR it's old code